mantidproject/mantid
# Abins atom-resolved spectrum

**Description of work.**

Modification/extension of the Abins algorithm to allow individual atom contributions to be selected. This is implemented with a prefix (`"atom_"` , set by AbinsModules.AbinsConstants.ATOM_PREFIX), using the existing API/GUI algorithm option "atoms". The existing behaviour is that "atoms" is a list of element names; under the modified behaviour elements and atoms may be mixed e.g. by entering "Si,atom_1,atom_2,Cl" in the corresponding GUI field. When Abins is used to calculate simulated S data, it returns a WorkspaceGroup including summations over Si and Cl atoms as well as spectra corresponding to the first and second atoms. These 1-indexed identifiers correspond to the atom order in the _ab initio_ input data file. This feature allows the contributions of inequivalent atomic sites to be examined even when they are the same isotope.

(*Update:* the underscore is now optional and should not be included in the ATOM_PREFIX variable. It will still be include din the workspace name.)

**To test:**
Unit tests and doctests have been updated.

To try out the changes, use strings as described above with sample data. The _squaricn.phonon_ file is suitable as it contains a variety of elements and inequivalent atomic sites; this is named "squaricn_sum_Abins.phonon" and is found (as an MD5 reference) under Testing/Data/UnitTest. The corresponding file in the data folder is MD5/189ef0c498b11e77aef83e5bc9940289, but it needs to be copied and renamed with an appropriate file extension (i.e. _*.phonon_) to be accepted by the Abins interface. GUI example below:
  
![abins_atomindex_example](https://user-images.githubusercontent.com/2511383/58643745-c8dd7380-82f7-11e9-85fb-6c36549c9c38.png)

This should run pretty quickly and give sensible errors if using a zero index or an unacceptable name (i.e. not a known element, atom_INT or atomINT).

*There is no associated issue.*

I'm not sure what the procedure is for adding release notes? A short summary of this change would be:
"Simulated INS spectra from Abins may be decomposed into specific atomic contributions, selected by their order in the imported _ab initio_ data. These selections are made in addition to the existing selection of element-wise contributions."

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
